scipy/stats/tests/test_stats.py

```diff
@@ -5965,226 +5960,223 @@ def test__broadcast_concatenate():
-    pr_2D = xp.asarray([pr, pr])
-
-    rvs3 = xp.linspace(1, 100, 25)
-    rvs2 = xp.linspace(1, 100, 100)
-    rvs1 = xp.linspace(5, 105, 100)
-    rvs1_2D = xp.stack([rvs1, rvs2])
-    rvs2_2D = xp.stack([rvs2, rvs1])
-
-    t, p = stats.ttest_ind(rvs1, rvs2, axis=0, equal_var=False)
-    xp_assert_close(t, tr)
-    xp_assert_close(p, pr)
-
-    t, p = stats.ttest_ind_from_stats(*_desc_stats(rvs1, rvs2), equal_var=False)
-    xp_assert_close(t, tr)
-    xp_assert_close(p, pr)
-
-    t, p = stats.ttest_ind(rvs1, rvs3, axis=0, equal_var=False)
-    xp_assert_close(t, tr_uneq_n)
-    xp_assert_close(p, pr_uneq_n)
-
-    t, p = stats.ttest_ind_from_stats(*_desc_stats(rvs1, rvs3), equal_var=False)
-    xp_assert_close(t, tr_uneq_n)
-    xp_assert_close(p, pr_uneq_n)
-
-    res = stats.ttest_ind(rvs1_2D.T, rvs2_2D.T, axis=0, equal_var=False)
-    xp_assert_close(res.statistic, tr_2D)
-    xp_assert_close(res.pvalue, pr_2D)
-
-    args = _desc_stats(rvs1_2D.T, rvs2_2D.T)
-    res = stats.ttest_ind_from_stats(*args, equal_var=False)
-    xp_assert_close(res.statistic, tr_2D)
-    xp_assert_close(res.pvalue, pr_2D)
-
-    res = stats.ttest_ind(rvs1_2D, rvs2_2D, axis=1, equal_var=False)
-    xp_assert_close(res.statistic, tr_2D)
-    xp_assert_close(res.pvalue, pr_2D)
-
-    args = _desc_stats(rvs1_2D, rvs2_2D, axis=1)
-    res = stats.ttest_ind_from_stats(*args, equal_var=False)
-    xp_assert_close(res.statistic, tr_2D)
-    xp_assert_close(res.pvalue, pr_2D)
-
-
-@pytest.mark.filterwarnings(
-    "ignore:divide by zero encountered:RuntimeWarning"
-) # for dask
-@pytest.mark.filterwarnings(
-    "ignore:invalid value encountered:RuntimeWarning"
-) # for dask
-@make_skip_xp_backends('ttests')
-def test_ttest_ind_zero_division(xp):
-    # test zero division problem
-    x = xp.zeros(3)
-    y = xp.ones(3)
-    with pytest.warns(RuntimeWarning, match="Precision loss occurred"):
-        t, p = stats.ttest_ind(x, y, equal_var=False)
-    xp_assert_equal(t, xp.asarray(-xp.inf))
-    xp_assert_equal(p, xp.asarray(0.))
-
-    with np.errstate(all='ignore'):
-        t, p = stats.ttest_ind(x, x, equal_var=False)
-        xp_assert_equal(t, xp.asarray(xp.nan))
-        xp_assert_equal(p, xp.asarray(xp.nan))
-
-        # check that nan in input array result in nan output
-        anan = xp.asarray([[1, xp.nan], [-1, 1]])
-        t, p = stats.ttest_ind(anan, xp.zeros((2, 2)), equal_var=False)
-        xp_assert_equal(t, xp.asarray([0., np.nan]))
-        xp_assert_equal(p, xp.asarray([1., np.nan]))
-
-
-def test_ttest_ind_nan_2nd_arg():
-    # regression test for gh-6134: nans in the second arg were not handled
-    x = [np.nan, 2.0, 3.0, 4.0]
-    y = [1.0, 2.0, 1.0, 2.0]
-
-    r1 = stats.ttest_ind(x, y, nan_policy='omit')
-    r2 = stats.ttest_ind(y, x, nan_policy='omit')
-    assert_allclose(r2.statistic, -r1.statistic, atol=1e-15)
-    assert_allclose(r2.pvalue, r1.pvalue, atol=1e-15)
+@make_skip_xp_backends(stats.ttest_ind)
+class TestTTestInd:
```

Comment:
Thank you for doing this. Should the stragglers above be moved over? And maybe it makes sense to have a separate class for combined tests rather than putting this first test in `TestTTestInd`. It's definitely a test of both, not a test of one against the other.